frame/benchmarking/src/lib.rs

```diff
@@ -764,120 +812,68 @@ macro_rules! impl_benchmark {
-						$crate::log::trace!(
-							target: "benchmark",
-							"Start Benchmark: {:?}", c
-						);
+					// Time the extrinsic logic.
+					$crate::log::trace!(
+						target: "benchmark",
+						"Start Benchmark: {:?}", c
+					);
 
-						let start_pov = $crate::benchmarking::proof_size();
-						let start_extrinsic = $crate::benchmarking::current_time();
+					let start_pov = $crate::benchmarking::proof_size();
+					let start_extrinsic = $crate::benchmarking::current_time();
 
-						closure_to_benchmark()?;
+					closure_to_benchmark()?;
 
-						let finish_extrinsic = $crate::benchmarking::current_time();
-						let end_pov = $crate::benchmarking::proof_size();
+					let finish_extrinsic = $crate::benchmarking::current_time();
+					let end_pov = $crate::benchmarking::proof_size();
 
-						// Calculate the diff caused by the benchmark.
-						let elapsed_extrinsic = finish_extrinsic.saturating_sub(start_extrinsic);
-						let diff_pov = match (start_pov, end_pov) {
-							(Some(start), Some(end)) => end.saturating_sub(start),
-							_ => Default::default(),
-						};
+					// Calculate the diff caused by the benchmark.
+					let elapsed_extrinsic = finish_extrinsic.saturating_sub(start_extrinsic);
+					let diff_pov = match (start_pov, end_pov) {
+						(Some(start), Some(end)) => end.saturating_sub(start),
+						_ => Default::default(),
+					};
 
-						// Commit the changes to get proper write count
-						$crate::benchmarking::commit_db();
-						$crate::log::trace!(
-							target: "benchmark",
-							"End Benchmark: {} ns", elapsed_extrinsic
-						);
-						let read_write_count = $crate::benchmarking::read_write_count();
-						$crate::log::trace!(
-							target: "benchmark",
-							"Read/Write Count {:?}", read_write_count
-						);
+					// Commit the changes to get proper write count
+					$crate::benchmarking::commit_db();
+					$crate::log::trace!(
+						target: "benchmark",
+						"End Benchmark: {} ns", elapsed_extrinsic
+					);
+					let read_write_count = $crate::benchmarking::read_write_count();
+					$crate::log::trace!(
+						target: "benchmark",
+						"Read/Write Count {:?}", read_write_count
+					);
 
-						// Time the storage root recalculation.
-						let start_storage_root = $crate::benchmarking::current_time();
-						$crate::storage_root();
-						let finish_storage_root = $crate::benchmarking::current_time();
-						let elapsed_storage_root = finish_storage_root - start_storage_root;
+					// Time the storage root recalculation.
+					let start_storage_root = $crate::benchmarking::current_time();
+					$crate::storage_root();
+					let finish_storage_root = $crate::benchmarking::current_time();
+					let elapsed_storage_root = finish_storage_root - start_storage_root;
 
-						let read_and_written_keys = $crate::benchmarking::get_read_and_written_keys();
+					let skip_meta = [ $( stringify!($name_skip_meta).as_ref() ),* ];
+					let read_and_written_keys = if (&skip_meta).contains(&extrinsic) {
```

Comment:
The compiler should do an autoref when calling `contains`?

```suggestion
					let read_and_written_keys = if skip_meta.contains(&extrinsic) {
```